neo4j/conf.py

```diff
@@ -242,6 +247,10 @@ def get_ssl_context(self):
         # https://docs.python.org/3.6/library/ssl.html#ssl.PROTOCOL_TLS_CLIENT
         ssl_context = ssl.SSLContext(ssl.PROTOCOL_TLS_CLIENT)
 
+        # For recommended security options see
+        # https://docs.python.org/3.10/library/ssl.html#protocol-versions
+        ssl_context.minimum_version = ssl.TLSVersion.TLSv1_2
```

Comment:
If you have a look at https://docs.python.org/3.10/library/ssl.html#ssl.SSLContext.minimum_version you'll find that `SSLContect.minimum_version` is 

> New in version 3.7.

Since the 4.4 driver must keep support for Python 3.6, we can't use this here or must use a condition (`if`) to find a solution pre Python 3.7 and post Python 3.7